ODIN_II/SRC/netlist_create_from_ast.cpp

```diff
@@ -1966,30 +1969,55 @@ void create_symbol_table_for_module(ast_node_t* module_items, STRING_CACHE_LIST
 						sc_spot = sc_lookup_string(local_symbol_table_sc, temp_string);
 						if( sc_spot == -1 )
 						{
-							sc_spot = sc_add_string(local_symbol_table_sc, temp_string);
-
-							/* store the data which is an idx here */
-							local_symbol_table_sc->data[sc_spot]= module_items->children[i]->children[0];
+							implicit_declarations = (ast_node_t **)vtr::realloc(implicit_declarations, sizeof(ast_node_t*)*(num_implicit_declarations+1));
+							implicit_declarations[0] = module_items->children[i]->children[0]->children[0];
+							num_implicit_declarations++;
+						}
+						vtr::free(temp_string);
+					}
+				}
+			}
+		}
 
-							/* store the symbol */
-							local_symbol_table = (ast_node_t **)vtr::realloc(local_symbol_table, sizeof(ast_node_t*)*(num_local_symbol_table+1));
-							local_symbol_table[num_local_symbol_table] = (ast_node_t *)module_items->children[i]->children[0];
-							num_local_symbol_table ++;
+		/* add implicit declarations to string cache */
+		for (i = 0; i < num_implicit_declarations; i++)
+		{
+			ast_node_t *node = implicit_declarations[i];
+			sc_spot = sc_add_string(local_symbol_table_sc, node->types.identifier);
+			oassert(sc_spot > -1);
 
+			if (local_symbol_table_sc->data[sc_spot] == NULL)
+			{
+				ast_node_t *var_declare = create_node_w_type(VAR_DECLARE, node->line_number, node->file_number);
+			
+				/* copy the output status over */
+				var_declare->types.variable.is_wire = true;
+				var_declare->types.variable.is_reg = false;
 
-							/* copy the output status over */
-							((ast_node_t*)local_symbol_table_sc->data[sc_spot])->types.variable.is_wire = true;
-							((ast_node_t*)local_symbol_table_sc->data[sc_spot])->types.variable.is_reg = false;
+				var_declare->types.variable.is_integer = false;
+				var_declare->types.variable.is_input = false;
+				
+				allocate_children_to_node(var_declare, { node, NULL, NULL, NULL, NULL, NULL });
 
-							((ast_node_t*)local_symbol_table_sc->data[sc_spot])->types.variable.is_integer = false;
-							((ast_node_t*)local_symbol_table_sc->data[sc_spot])->types.variable.is_input = false;
+				/* store the data which is an idx here */
```

Comment:
I would change the comment since you are storing a node, can be misleading. Do you mean an index for and array ref ast node?

Reply:
this is copy/pasted from everything else in this function that stores a node in the string cache. it's doing the same thing, storing a var_declare (data) for lookups later, nothing to do with array refs

Reply:
i would simply remove the comment if you don't mind since it makes no sense there